modules/core/src/main/java/org/locationtech/jts/io/WKBReader.java

```diff
@@ -202,7 +202,6 @@ else if(isStrict)
 
     // determine if SRIDs are present
     hasSRID = (typeInt & 0x20000000) != 0;
-    int SRID = 0;
```

Comment:
So that we use the class field. This might be problematic in case a geometry collection contains mixed SRIDs as @dr-jts pointed [here](https://github.com/locationtech/jts/issues/663#issuecomment-759626055).

Reply:
Yes, definitely problematic.  This should just use the SRID of the top-most geometry (or 0 if not set).  JTS does not support mixed SRIDs in collections.

Reply:
Shouldn't we throw a `ParseException` exception in that case instead of using the SRID of top-most geometry?

Reply:
It's nice to allow reading as many WKB inputs as possible.   Even if they are invalid, they can only be fixed if they are able to be read.

To preserve the most information, the SRIDs should always be read, and then applied to any missing SRIDs in child geometries.  This will require passing the most recent SRID down the call stack as a parameter.

Reply:
Added a commit, PTAL.

Reply:
Looks good.